zombies.js

```diff
@@ -79,136 +96,236 @@
+   * Player equips a weapon item.
+   *
+   * Player can only equip Weapon instances.
+   * Player can only equip weapon items from their pack.
+   *
+   * If the player already has a weapon equipped (the equipped property
+   *   is set to an Item), find the itemToEquip in the pack and replace
+   *   it with the currently equipped item.  Then set the equipped property
+   *   to the itemToEquip.
+   * However, if the player doesn't already have a weapon equipped, simply
+   *   equip that item and remove it from the pack.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name equip
+   * @param {Weapon} itemToEquip  The weapon item to equip.
+   */
+
+  equip(itemToEquip) {
+    let index = this._pack.indexOf(itemToEquip);
+
+    if (itemToEquip instanceof Weapon) {
+      if (index === -1) {
+        console.log('You dont own that weapon');
+      } else if (!this.equipped && index > -1) {
+        this._pack.splice(index, 1);
+        this.equipped = itemToEquip;
+      } else {
+        this._pack.splice(index, 1, this.equipped);
+        this.equipped = itemToEquip;
+      }
+    } else {
+      console.log("You can't equip that");
+    }
+  };
+
+  /**
+   * Player Class Method => eat(itemToEat)
+   * -----------------------------
+   * Player eats a food item, restoring their health.
+   *
+   * Player can only eat Food instances.
+   * Player can only eat food items from their pack.
+   *
+   * Remove itemToEat from the pack.
+   * Increase the player's health by the food's energy amount, but do not
+   *   exceed the player's max health.  If exceeded, simply set player's health
+   *   to max health instead.
+   * To access the player's max health, be sure to use Player's getMaxHealth method.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name eat
+   * @param {Food} itemToEat  The food item to eat.
+   */
+
+  eat(itemToEat) {
+    if (itemToEat instanceof Food) {
+      let index = this._pack.indexOf(itemToEat);
+      let maxHealth = this.getMaxHealth();
+      if (index === -1) {
+        console.log(`You don't have a(n) ${itemToEat}`);
+      } else {
+        this._pack.splice(index, 1);
+        if (this.health + itemToEat.energy > maxHealth) {
+          this.health = maxHealth;
+        } else {
+          this.health += itemToEat.energy;
+        }
+      }
+    } else {
+      console.log('You cant eat that, moron');
+    }
+  };
+
+  /**
+   * Player Class Method => useItem(item)
+   * -----------------------------
+   * Player uses an item from the pack.
+   *
+   * If the item is a weapon, the player should equip the item.
+   * If the item is food, the player should eat the item.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name useItem
+   * @param {Item/Weapon/Food} item   The item to use.
+   */
+
+  useItem(item) {
+    if (item instanceof Food) {
+      this.eat(item);
+    } else {
```

Comment:
There should be a check to ensure that you're getting a weapon before you equip something.